netbox/extras/events.py

```diff
@@ -119,6 +120,22 @@ def process_event_rules(event_rules, model_name, event, data, username=None, sna
             script_name = event_rule.action_parameters['script_name']
             script = script_module.scripts[script_name]()
 
+            # Process Action Data
+
+            if event_rule.action_data:
+                context = {
+                    'event': event,
+                    'timestamp': timezone.now().isoformat(),
+                    'model': model_name,
+                    'username': username,
+                    'request_id': request_id,
+                    'model': data,
+                }
+                rendered_data = render_jinja2(json.dumps(event_rule.action_data), context)
```

Comment:
Sorry, I don't see why the render_jinja2 is needed here - event_rule.action_data should just be the json (dictionary) which should be able to get directly passed to the script.  I'm thinking you could just change the Job.enqueue to pass event_rule.action_data as the data param.

Reply:
@arthanson  If we don't render the jinja2 how the user gonna pass variables to the script, for example:
1 - For example the Script form expects an Interface ID, or some model attribute.
If It didn't run the jinja2 how the user gonna be able to pass models variables to be processed by the script form?

Reply:
The user doens't have the model params when he defines the event rule, so I think there must be some way for him to reference the models and the changelogs, to build the data that gonna be passed to the Script for

Reply:
The `action_data` field on EventRule holds a dictionary of arbitrary additional data. This doesn't require any mutation; it just needs to be passed to the script, and the script needs to accommodate whatever data is being passed as variables.

I think you're trying to accomplish something different than what I agreed to [here](https://github.com/netbox-community/netbox/issues/14884#issuecomment-1943820352). There's no reason to manipulate the data before sending it to the script; the script just needs to be written to accept the data as passed.

Reply:
@jeremystretch even if I pass the `event_data` as is to the script how the Script developer gonna be able to reference the event object like the changed object, since it doesn't have that data inside the Script? another thing is that way breaks the common script logic where the developer expects the data dict to return a parsed form

Reply:
Going further if the action_data is a static value what's the meaning of having it at all, the Script dev could just create the script with that data hard coded